encoding/cloner.go

```diff
@@ -0,0 +1,29 @@
+package encoding
+
+import (
+	refmt "github.com/polydawn/refmt"
+	"github.com/polydawn/refmt/obj/atlas"
+)
+
+// PooledCloner is a thread-safe pooled object cloner.
+type PooledCloner struct {
+	Count   int
+	cloners chan refmt.Cloner
```

Comment:
is this better than a sync.Pool?

Reply:
I was under the impression that the elements in a pool get GCed every GC cycle. However, it does appear faster in the benchmarks and we really care more about throughput than latency so I've switched to pools.